x-pack/plugins/ml/public/application/jobs/new_job/utils/new_job_utils.test.ts

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { IUiSettingsClient } from 'kibana/public';
+import { IIndexPattern } from '../../../../../../../../src/plugins/data/common/index_patterns';
+import { SavedSearchSavedObject } from '../../../../../common/types/kibana';
+import { createSearchItems } from './new_job_utils';
+
+describe('createSearchItems', () => {
+  const savedSearch = ({
+    client: {
+      http: {
+        basePath: {
+          basePath: '/abc',
+          serverBasePath: '/abc',
+        },
+        anonymousPaths: {},
+      },
+      batchQueue: [],
+    },
+    attributes: {
+      title: 'not test',
+      description: '',
+      hits: 0,
+      columns: ['_source'],
+      sort: [],
+      version: 1,
+      kibanaSavedObjectMeta: {
+        searchSourceJSON: '',
+      },
+    },
+    _version: 'WzI0OSw0XQ==',
+    id: '4b9b1010-c678-11ea-b6e6-e942978fa29c',
+    type: 'search',
+    migrationVersion: {
+      search: '7.4.0',
+    },
+    references: [
+      {
+        name: 'kibanaSavedObjectMeta.searchSourceJSON.index',
+        type: 'index-pattern',
+        id: '7e252840-bd27-11ea-8a6c-75d1a0bd08ab',
+      },
+    ],
+  } as unknown) as SavedSearchSavedObject;
+
+  const kibanaConfig = {} as IUiSettingsClient;
+  const indexPattern = ({
+    fields: [],
+  } as unknown) as IIndexPattern;
+
+  test('should match index pattern', () => {
+    const resp = createSearchItems(kibanaConfig, indexPattern, null);
+    expect(resp).toStrictEqual({
+      combinedQuery: { bool: { must: [{ match_all: {} }] } },
+      query: { query: '', language: 'lucene' },
+    });
+  });
+
+  test('should match saved search with kuery and condition', () => {
+    const searchSource = {
+      highlightAll: true,
+      version: true,
+      query: { query: 'airline : "AAL" ', language: 'kuery' },
+      filter: [],
+      indexRefName: 'kibanaSavedObjectMeta.searchSourceJSON.index',
+    };
+    savedSearch.attributes.kibanaSavedObjectMeta.searchSourceJSON = JSON.stringify(searchSource);
```

Comment:
as you mutate `savedSearch` in each test, it's better to initialize it in `beforeEach` callback to avoid potential issues

Reply:
changed in 41db653